pkg/provision/primitives/cache/cache.go

```diff
@@ -115,7 +171,11 @@ func (s *Fs) Add(r *provision.Reservation) error {
 		return err
 	}
 
-	f, err := os.OpenFile(filepath.Join(s.root, r.ID), os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0660)
+	flags := os.O_CREATE | os.O_WRONLY
+	if !update {
+		flags |= os.O_EXCL
+	}
```

Comment:
```suggestion
	flags :=  os.O_WRONLY
	if !update {
		flags |= os.O_EXCL | os.O_CREATE
	}
```
Seems you only need the O_CREATE flag only when you do NOT update, right ?

Reply:
Do we care? create will create it if the file does not exist, which update should still do if the file is gone for any reason.

Reply:
the previous behavior was we return an error if there was already a reservation present. But I guess you are write in both case we should create it. so ignore me :-P